### PR TITLE
FAQ additions

### PR DESCRIPTION
**Pull request by purplem1lk**
# PR Description

This PR fixes the issue number: 

### Summary of my changes and explanation of my decisions

- Added FAQs received from users.
- Added product information to help differentiate between our technologies and products at a high level.

### Self-check

Please check all that apply:

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my code or content update and edited it to the best of my abilities
- [ ] I have commented my code, particularly in hard-to-understand areas; my comments are concise

### Contributors list

Would you like your contribution to be celebrated? Please check all that apply:

- [ ] I would like to be featured on the future contributors list in the README. If so, please tell us:
  - what name you'd like to be shown there?
  - we usually link github profile pic -- is that ok? If not, provide another url: 
  - we usually link your github profile but if you have a portfolio page, provide a link:

- [ ] I would like to get a shoutout in the next monthly updates post. If so, please provide your twitter handle (or we will link to your GitHub profile).

⚡️ ⚡️ ⚡️  Thank you for contributing to StackBlitz ⚡️ ⚡️ ⚡️ 

-----
<a href="https://stackblitz.com/~/github.com/purplem1lk/docs/tree/main">Review PR in StackBlitz Codeflow</a> _Submitted with [StackBlitz Codeflow](https://stackblitz.com/~/github.com/purplem1lk/docs/tree/main)._

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204252713773858

### CODE REVIEW

**Comment by purplem1lk** [comment]
@sylwiavargas @kc0tlh additional FAQs added. Still WIP, but wanted to get an MVP up for guidance before adding further content

File: docs/guides/user-guide/general-faqs.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+We do offer [WebContainer API](webcontainers.io) licenses for organizations wanting to use WebContainer technology for production usage in a for-profit setting. Check out [webcontainers.io](webcontainers.io) for more information!
+
+# Technology FAQs
+
+## Classic Editor (Engineblock projects)
+
+### What is the Classic Editor for Engineblock projects?
+
+The Classic Editor utilizes Engineblock and WeBcontainer technologies to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```
Comment (purplem1lk) [inline docs/guides/user-guide/general-faqs.md:99]: Rewrite sentence - interpretted as both technologies are needed to run.

File: docs/guides/user-guide/general-faqs.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+Engineblock projects are distinguishable as they do not have a terminal.
```
Comment (purplem1lk) [inline docs/guides/user-guide/general-faqs.md:101]: distinguishable FROM webcontainers-based projects. for more info re: webcontainers, <ADD LINK that goes down to webcontainer FAQ block>

File: docs/guides/user-guide/general-faqs.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+The current TypeScript starter uses EngineBlock technology, which has a fixed TypeScript version. EngineBlock is great for lightweight frontends, but does not run Node.js itself and will be limited to things like TypeScript versioning.
+
+Webcontainer, however does run Node.js itself in the browser and does not have this limitation. You can simply `npm install` any TypeScript version as you would locally.
+
+## Classic Editor (WebContainer projects)
+
+### What is the Classic Editor for WebContainer projects?
+
+The Classic Editor utilizes Engineblock and WeBcontainer technologies to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```
Comment (purplem1lk) [inline docs/guides/user-guide/general-faqs.md:119]: rewrite sentence - reads as if BOTH technologies are needed to run classic editor, but not true.

File: docs/guides/user-guide/general-faqs.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+### Can I develop ElectronJS apps on StackBlitz?
+
+Since Electron is a native app, this cannot be supported since StackBlitz runs entirely within the browser.
+
+## Codeflow Editor
+
+### What is the Codeflow Editor?
+
+The Codeflow Editor utilizes our Codeflow technology to bring VSC code into the browser. This editor is capable of running full development workflows, including pushing commits and creating pull requests to GitHub. This bypasses the need to clone a repository or install dependencies on your local machine. If you haven't already, you can check out Codeflow [here](https://stackblitz.com/codeflow)!
```
Comment (purplem1lk) [inline docs/guides/user-guide/general-faqs.md:151]: rewrite as: "The Codeflow Editor brings VS Code into the browser."

Comment (purplem1lk) [reply docs/guides/user-guide/general-faqs.md:151]: after "...creating pull requests to GitHub", add the features as a separate sentence.

File: docs/guides/user-guide/general-faqs.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+- Text or formatting changes in documentation or blogs
+- Quick fixes that don't require dependency or compilation changes
+
+You can learn more about Web Publisher [here](https://developer.stackblitz.com/codeflow/content-updates-with-web-publisher).
+
+## Classic Editor SDK
+
+### What is the Classic Editor SDK?
+
+Our SDK (software development kit) allows users to create StackBlitz projects to be opened in a new window or embedded in your docs, example pages, or blog posts. Projects are controlled by using the SDK's VM (virtual machine) interface. The VM can be used to control the UI of an embedded StackBlitz project, change the currently open file(s), and read and write files frmo the project's virtual filesystem.
```
Comment (purplem1lk) [inline docs/guides/user-guide/general-faqs.md:179]: Our JavaScript-based SDK..... allows users to created SB Classic Editor projects.

Remove VM info. "The SDK can be used to control the UI......"

File: docs/guides/user-guide/general-faqs.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+We do offer [WebContainer API](webcontainers.io) licenses for organizations wanting to use WebContainer technology for production usage in a for-profit setting. Check out [webcontainers.io](webcontainers.io) for more information!
+
+# Technology FAQs
+
+## Classic Editor (Engineblock projects)
+
+### What is the Classic Editor for Engineblock projects?
+
+Different projects that use the Classic Editor utilize Engineblock technology to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:99]:
```suggestion
Different projects that use the Classic Editor utilize [EngineBlock](/guides/user-guide/available-environments#engineblock) technology to run. As the original StackBlitz UI, the Classic Editor has a file tree, an npm package manager, and a lightweight dev server and preview pane that allows you to see your app as you code.
```

File: docs/guides/user-guide/general-faqs.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+Engineblock projects are distinguishable from WebContainer-based projects as they do not have a terminal.
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:101]:
```suggestion
EngineBlock projects are distinguishable from WebContainers-based projects as they do not have a terminal.
```

File: docs/guides/user-guide/general-faqs.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+
+### How do I create code snippets?
+
+Code snippets are available in our Classic Editor for both Engineblock and WebContainer projects. Click on the Settings icon in the activity bar to the left of your screen. The Settings sidebar will pop open and you will see User snippets and Workspace snippets. User snippets are available for any of your projects. Workspace snippets are only available in the current project you create them in.
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:106]:
```suggestion
Code snippets are available in our Classic Editor for both EngineBlock and WebContainer projects. Click on the Settings icon in the activity bar to the [left of your screen](/guides/user-guide/ide-whats-on-your-screen#frontmatter-title). The Settings sidebar will pop open and you will see User snippets and Workspace snippets. User snippets are available for any of your projects. Workspace snippets are only available in the current project you create them in.
```

File: docs/guides/user-guide/general-faqs.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+
+### Can I change the TypeScript version in my project?
+
+The current TypeScript starter uses EngineBlock technology, which has a fixed TypeScript version. EngineBlock is great for lightweight frontends, but does not run Node.js itself and will be limited to things like TypeScript versioning.
+
+Webcontainer, however does run Node.js itself in the browser and does not have this limitation. You can simply `npm install` any TypeScript version as you would locally.
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:113]:
```suggestion
WebContainers, however, do run Node.js itself in the browser and do not have this limitation. You can simply `npm install` any TypeScript version as you would locally.
```

File: docs/guides/user-guide/general-faqs.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+## Classic Editor (WebContainer projects)
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:115]:
```suggestion
## Classic Editor (WebContainers projects)
```

File: docs/guides/user-guide/general-faqs.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+
+### What is the Classic Editor for WebContainer projects?
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:117]:
```suggestion
### What is the Classic Editor for WebContainers projects?
```

File: docs/guides/user-guide/general-faqs.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+
+Different projects that use the Classic Editor utilize WebContainer technology to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```
Comment (sulco) [inline docs/guides/user-guide/general-faqs.md:119]:
```suggestion
Different projects that use the Classic Editor utilize WebContainers technology to run. As the original StackBlitz UI, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```

File: docs/guides/user-guide/general-faqs.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+  - ['meta', {property: 'og:image', content: 'https://developer.stackblitz.com/img/og/general-faqs.png'}]
+  - ['meta', {name: 'twitter:title', content: *title}]
+  - ['meta', {name: 'twitter:description', content: *description}]
+---
+
+This page outlines commonly asked questions and answers about StackBlitz.
+
+# General FAQs
+
+Here's a [handy video](https://twitter.com/sulco/status/1275039159611203585) featured on Twitter that shows you how to create and use code snippets in StackBlitz.
```
[the remark this answers is not in the document]
Comment (purplem1lk) [reply docs/guides/user-guide/general-faqs.md]: Yes! Thank you for catching this. Added the question into my updated commit. Thanks!

File: docs/guides/user-guide/general-faqs.md [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+### What are the deploy options available on StackBlitz? 
+
+The three pathways we current support are:
+
+1. If you are in any non-Codeflow StackBlitz project, you can download the project to your local machine by clicking the button right underneath the "Share" dialogue and deploy manually.
+
+2. If you are using an EngineBlock project, you can deploy directly via our [Firebase integration](https://developer.stackblitz.com/guides/user-guide/ide-whats-on-your-screen#firebase-sidebar).
+
+3. If you are using Codeflow, you can push code directly into GitHub and then use the [Netlify](https://github.com/apps/netlify) or [Vercel](https://vercel.com/docs/concepts/git/vercel-for-github) GitHub bots to deploy your site.
+
```
[the remark this answers is not in the document]
Comment (purplem1lk) [reply docs/guides/user-guide/general-faqs.md:36]: Thanks for pointing this out! Added this to my latest commit.

**Review by sulco — APPROVED** [review]

**Pull request merged**